ffi/derive/src/convert.rs

```diff
@@ -0,0 +1,445 @@
+                source: <Self as iroha_ffi::TryFromReprC<'itm>>::Source,
+                _: &mut <Self as iroha_ffi::TryFromReprC<'itm>>::Store
+            ) -> iroha_ffi::Result<Self> {
+                #( #discriminants )*
+
+                unsafe { match *source {
+                    #( | #discriminant_names )* => Ok(&mut *(source as *mut _ as *mut _)),
+                    _ => Err(iroha_ffi::FfiReturn::TrapRepresentation),
+                }}
+            }
+        }
+
+        impl<'slice> iroha_ffi::slice::TryFromReprCSliceRef<'slice> for #enum_name {
+            type Source = iroha_ffi::slice::SliceRef<'slice, Self>;
+            type Store = ();
+
+            unsafe fn try_from_repr_c(
+                source: <Self as iroha_ffi::slice::TryFromReprCSliceRef<'slice>>::Source,
+                _: &mut <Self as iroha_ffi::slice::TryFromReprCSliceRef<'slice>>::Store
+            ) -> iroha_ffi::Result<&'slice [Self]> {
+                source.into_rust().ok_or(iroha_ffi::FfiReturn::ArgIsNull)
+            }
+        }
+        impl<'slice> iroha_ffi::slice::TryFromReprCSliceMut<'slice> for #enum_name {
+            type Source = iroha_ffi::slice::SliceMut<'slice, #enum_name>;
+            type Store = ();
+
+            unsafe fn try_from_repr_c(
+                source: <Self as iroha_ffi::slice::TryFromReprCSliceMut<'slice>>::Source,
+                _: &mut <Self as iroha_ffi::slice::TryFromReprCSliceMut>::Store
+            ) -> iroha_ffi::Result<&'slice mut [Self]> {
+                source.into_rust().ok_or(iroha_ffi::FfiReturn::ArgIsNull)
+            }
+        }
+    }
+}
+
+fn derive_into_ffi_for_opaque_item_wrapper(name: &Ident) -> TokenStream2 {
+    let opaque_item_slice_into_ffi_derive = derive_into_ffi_for_opaque_item_slice(name);
+
+    quote! {
+        impl iroha_ffi::IntoFfi for #name {
+            type Target = *mut iroha_ffi::Opaque;
+
+            fn into_ffi(self) -> Self::Target {
+                core::mem::ManuallyDrop::new(self).__opaque_ptr
+            }
+        }
+
+        impl iroha_ffi::IntoFfi for &#name {
+            type Target = *const iroha_ffi::Opaque;
+
+            fn into_ffi(self) -> Self::Target {
+                self.__opaque_ptr
+            }
+        }
+
+        impl iroha_ffi::IntoFfi for &mut #name {
+            type Target = *mut iroha_ffi::Opaque;
+
+            fn into_ffi(self) -> Self::Target {
+                self.__opaque_ptr
+            }
+        }
+
+        #opaque_item_slice_into_ffi_derive
+    }
+}
+
+fn derive_into_ffi_for_opaque_item(name: &Ident) -> TokenStream2 {
+    let opaque_item_slice_into_ffi_derive = derive_into_ffi_for_opaque_item_slice(name);
+
+    quote! {
+        impl iroha_ffi::IntoFfi for #name {
+            type Target = *mut Self;
+
+            fn into_ffi(self) -> Self::Target {
+                let layout = core::alloc::Layout::for_value(&self);
+
+                unsafe {
+                    let ptr: Self::Target = alloc(layout).cast();
```

Comment:
Because of this we had to import `alloc` in every crate that uses `IntoFfi`, right?
And we can't depend on anykind of conditional compilation here becasue we can't know feature that turns on `std` beforehand.

Reply:
Sadly yes. Fortunately, this is going to get fixed very soon.